packages/ui/src/components/alert/AlertRoot.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Children,
+  forwardRef,
+  isValidElement,
+  PropsWithChildren,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+
+import { Button } from '@components/button'
+import { Icon, IconNameMap } from '@components/icon'
+import { useResizeObserver } from '@hooks/use-resize-observer'
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+import { AlertTitle } from './AlertTitle'
+
+const alertVariants = cva('cn-alert', {
+  variants: {
+    theme: {
+      info: 'cn-alert-info',
+      danger: 'cn-alert-danger',
+      warning: 'cn-alert-warning'
+    }
+  },
+  defaultVariants: {
+    theme: 'info'
+  }
+})
+
+const iconMap: Record<NonNullable<VariantProps<typeof alertVariants>['theme']>, keyof typeof IconNameMap> = {
+  info: 'info-circle',
+  danger: 'cross-circle',
+  warning: 'warning-triangle-outline'
+}
+
+const MAX_HEIGHT_WITH_TITLE = 70
+const MAX_HEIGHT_WITHOUT_TITLE = 60
+
+export interface AlertRootProps extends PropsWithChildren<VariantProps<typeof alertVariants>> {
+  className?: string
+  dismissible?: boolean
+  onDismiss?: () => void
+  expandable?: boolean
+}
+
+export const AlertRoot = forwardRef<HTMLDivElement, AlertRootProps>(
+  ({ className, theme, children, dismissible, onDismiss, expandable }, ref) => {
+    const [isVisible, setIsVisible] = useState(true)
+    const [isExpanded, setIsExpanded] = useState(false)
+    const [isOverflowing, setIsOverflowing] = useState(false)
+    const contentRef = useRef<HTMLDivElement>(null)
+
+    const handleDismiss = () => {
+      setIsVisible(false)
+      onDismiss?.()
+    }
+
+    const iconName: keyof typeof IconNameMap = iconMap[theme ?? 'info']
+
+    const toggleExpand = () => setIsExpanded(prev => !prev)
+
+    const shouldShowButton = expandable && isOverflowing
+
+    const hasTitle = Children.toArray(children).some(
+      (child: React.ReactNode) => isValidElement(child) && child?.type === AlertTitle
```

Comment:
```suggestion
      (child: ReactNode) => isValidElement(child) && child?.type === AlertTitle
```
Avoid using types that are not imported from React - it can lead to type errors when components/views are imported from UI to other apps.